hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -3174,9 +3175,8 @@ protected void insert(
 									.executeUpdate( insert ), insert, -1
 					);
 				}
-
 			}
-			catch (SQLException e) {
+			catch (SQLException | JDBCException e) {
```

Comment:
The added `JDBCException` should only be needed if an application provides its own `BatchBuilder` that builds a `Batch` that implements `#addToBatch` that does not call `#abortBatch` when it fails.

`JtaWithFailingBatchTest` throws `JDBCException`, which is handled here.

FWIW, there are lots of other places where this could happen.